app/src/renderer/components/staking/PageValidator.vue

```diff
@@ -392,8 +392,12 @@ export default {
     }
   },
   watch: {
-    validator(validator) {
-      this.$store.dispatch(`getSelfBond`, validator)
+    validator: {
+      immediate: true,
+      handler(validator) {
+        if (!validator) return
+        this.$store.dispatch(`getSelfBond`, validator)
```

Comment:
shouldn't we use `mapActions` instead?

Reply:
We haven't discussed a rule for this yet. Did it anyway to push things along.

Reply:
changed it but then e2e tests failed so I reverted it back to not blow the scope.

Reply:
ok we can discuss it later